test/azure/scanner.spec.ts

```diff
@@ -1,24 +1,32 @@
 import {assert} from 'chai';
 import {Scanner} from '../../src/scanner';
-import {subscriptionId} from '.';
+import {resourceGroup, subscriptionId} from '.';
+import {ResourceGraphTarget, RuleType} from '../../src/rules';
 
 describe('Scanner', function () {
   this.slow(5000);
   this.timeout(8000);
   it('can load and execute resource graph rules from a JSON file', async () => {
+    const rgTarget: ResourceGraphTarget = {
+      type: RuleType.ResourceGraph,
+      subscriptionId,
+    };
     const scanner = new Scanner();
     await scanner.loadRulesFromFile('../test/rules.json');
-    const rgResults = await scanner.scan('resourceGraph', subscriptionId);
-    assert.equal(rgResults.length, 1);
+    const rgResults = await scanner.scan(rgTarget);
+    const re = new RegExp(`/resourceGroups/${resourceGroup}`);
+    assert.isAtLeast(rgResults.length, 2);
     rgResults.forEach(r => {
-      assert.containsAllKeys(r, ['ruleName', 'description', 'total', 'ids']);
-      assert.equal(r.total, 0);
-    });
-    const dummyResults = await scanner.scan('dummy', '');
-    assert.equal(dummyResults.length, 2);
-    dummyResults.forEach(r => {
-      assert.containsAllKeys(r, ['ruleName', 'description', 'total', 'ids']);
-      assert.equal(r.total, 0);
+      assert.containsAllKeys(r, [
+        'ruleName',
+        'description',
+        'total',
+        'resourceIds',
+      ]);
+      if (r.ruleName === 'get-vnets') {
```

Comment:
More really specific assertions here that should probably be its own test case (or something else)

Our test is "can load and execute resource graph rules from a JSON file", ie: "Can I run **any** rule?" Seems like if we successfully pass `scanner.scan()` and get a result object w/o error, then we have some confidence that is true

Here, everything below that line seems to be all about asserting details of the response specifically for the `get-vnets` rule, so I think you're actually checking for something else in the same test case

Reply:
Good point. That makes sense